this has literally never broken, which is awesome. However, I haven't touched it since making, which was long before good error handling, but man it's been damn god since making it.

i forgot this grabs *every* location, not just nearby locations. Changed the func to only get all users locations. issue solved